src/core/defs.jl

```diff
@@ -132,17 +132,18 @@ description(def::DatumDef) = def.description
 
 unit(def::DatumDef) = def.unit
 
-# step_size(md::ModelDef) = step_size(indexvalues(md))
+function start_step(md::ModelDef)
+    keys::Vector{Int} = dim_keys(md, :time) # keys are, e.g., the start_times of the model runs
+    return keys[1], (length(keys) > 1 ? keys[2] - keys[1] : 1)
+end
 
-function step_size(md::ModelDef)
-    # N.B. assumes that all timesteps of the model are the same length
-    keys::Vector{Int} = dim_keys(md, :time) # keys are, e.g., the years the model runs
-    return length(keys) > 1 ? keys[2] - keys[1] : 1
+function start_step(values::Vector{Int})
+     return values[1], (length(values) > 1 ? values[2] - values[1] : 1)
 end
 
-function step_size(values::Vector{Int})
-    # N.B. assumes that all timesteps of the model are the same length
-     return length(values) > 1 ? values[2] - values[1] : 1
+function starttimes(md::ModelDef)
```

Comment:
The function `starttimes` is redundant with `timelabels(md::ModelDef)`, and I think the latter is a clearer name. I note that `timelabels` isn't currently used, and on further reflection, should be renamed `time_labels`.